### ISSUE TITLE
atom.Parser atom:content type html might not be wrapped in DIV

### ISSUE THREAD

**Issue by terinjokes** [mmcdole/gofeed#40]
### Expected behavior

The following example entry, included within a valid Atom feed, should create a gofeed.Item with the following content.

```xml
<atom:entry>
  <atom:title>Parsing Atom with gofeed</atom:title>
  <atom:link href="https://example.com/blog/2016/04/18/parsing-atom-with-gofeed" />
  <atom:updated>2016-04-18T00:00:00+00:00</atom:updated>
  <atom:id>https://example.com/blog/2016/04/18/parsing-atom-with-gofeed</atom:id>
  <atom:content type="html">
    &lt;p&gt;This is a directly included child element, no wrapping in a DIV element.&lt;/p&gt;

    &lt;div class="not-root"&gt;&lt;p&gt;This DIV is part of the post content, wholly unrelated to what RFC 4287 might say about DIVs.&lt;/p&gt;&lt;/div&gt;
  </atom:content>
</atom:entry>
```

```go
for _, item := range feed.Items {
  fmt.Println(item.Content)
}
// <p>This is a directly included child element, no wrapping in a DIV element.</p>\n\n<div class="not-root"><p>This DIV is part of the post content, wholly unrelated to what RFC 4287 might say about DIVs.</p></div>
```

### Actual behavior

```go
for _, item := range feed.Items {
  fmt.Println(item.Content)
}
// <p>This DIV is part of the post content, wholly unrelated to what RFC 4287 might say about DIVs.</p>
```

### Steps to reproduce the behavior

The problematic feed is https://terinstock.com/atom.xml. The author is alright.

### Supporting documentation

[RFC 4287 § 4.1.3.3](https://tools.ietf.org/html/rfc4287#section-4.1.3.3):

> ```
    2.  If the value of "type" is "html", the content of atom:content
        MUST NOT contain child elements and SHOULD be suitable for
        handling as HTML.  The HTML markup MUST be escaped; for
        example, "<br>" as "&lt;br>".  The HTML markup SHOULD be such
        that it could validly appear directly within an HTML <DIV>
        element.  Atom Processors that display the content MAY use the
        markup to aid in displaying it.
    3.  If the value of "type" is "xhtml", the content of atom:content
        MUST be a single XHTML div element [XHTML] and SHOULD be suitable
        for handling as XHTML.  The XHTML div element itself MUST NOT be
        considered part of the content.  Atom Processors that display the
        content MAY use the markup to aid in displaying it.  The escaped
        versions of characters such as "&" and ">" represent those
        characters, not markup.
```

Of course, a DIV is valid within a DIV, but it's not required for type html. Even if the content was wrapped with a DIV, it should be considered part of the content, for the html type.

**Comment by mmcdole** [comment]
@terinjokes thanks for the well written and sourced issue!    I'll look at this later tonight.

**Comment by terinjokes** [comment]
@mmcdole Removing this OR case in [`parseAtomText`](https://github.com/mmcdole/gofeed/blob/59dbcc52906a0bbfc8619065df23dcc33e3c39ec/atom/parser.go#L675) resolve the reported issue, but I'm sure it's included to fix something else.

**Comment by mmcdole** [comment]
@terinjokes I've got a PR up for this, but I have a test case failing.  However, after reading the quoted part of the Atom spec that you provided... I think the test case may be wrong (this test case was ported from https://github.com/kurtmckee/feedparser)

Test: https://github.com/mmcdole/gofeed/blob/atom_html_div_handling/testdata/parser/atom/atom10_feed_entry_content_div_escaped_markup.xml
Test Expected Result: https://github.com/mmcdole/gofeed/blob/atom_html_div_handling/testdata/parser/atom/atom10_feed_entry_content_div_escaped_markup.json

**Comment by terinjokes** [comment]
If it were me, I'd opt for the second behavior, leaving the <div> in if it's in the HTML content type. Stripping is explicitly mentioned in the XHTML case with no mention made in the HTML case.

A middle ground could be to unwrap the <div> if it's the only root `html.Node`. Would resolve this use case, and still account for other incorrect feeds.

**Comment by mmcdole** [comment]
I've merged #42 into master.

After thinking on it some, I decided on the following compromise:

- If you are within spec (encoded html only) and your type is `html`, no div/element will be removed.
- If you are out of spec, by having an unencoded wrapper div with no siblings, it will be removed even if your type is `html`.  

My goal ultimately is to parse the feeds as faithfully to the spec as a can. I don't want any change I introduce to modify a well-formed feed in a way that the spec does not say to.  However, if you step outside the spec, I'm going to take a little latitude with doing my best to correct the users inevitable mistakes.
